src/libraries/System.Text.Json/src/System/Text/Json/Serialization/Converters/Object/ObjectDefaultConverter.cs

```diff
@@ -152,10 +153,10 @@ internal override bool OnTryRead(ref Utf8JsonReader reader, Type typeToConvert,
                         // Read method would have thrown if otherwise.
                         Debug.Assert(tokenType == JsonTokenType.PropertyName);
 
+                        ReadOnlySpan<byte> unescapedPropertyName = JsonSerializer.GetPropertyName(ref state, ref reader, options);
```

Comment:
You pulled this out from `LookupProperty` to pass less state?

Reply:
Yes I should have commented on this in the PR. It is faster this way (to pass a couple less variables and return the value).